### DISCUSSION TITLE
Not able to download github file using raw link

### DISCUSSION THREAD

**Discussion by gaurav-soruppu** [mrdbourke/pytorch-deep-learning#725] [Q&A]
When i try to download the files from git using the requests package, it times out and is unable to download the file. I checked the code and its not an issue with the code. It has something to do with the raw URL. Is there another workaround to this?
For now i have been manually downloading and replacing the files.

**Comment by shisirkha** [comment]
use this code if yours's not work. it's probably problem with your link.

```
#Let us use some helper_functions from other
import requests
from pathlib import Path
import time
if Path("helper_functions.py").is_file():
  print(f"Already have this helper_functions.py")
  #import that
  from helper_functions import *
else:
  print("downloading the file...")
  time.sleep(2)
  r = requests.get('https://github.com/mrdbourke/pytorch-deep-learning/raw/main/helper_functions.py')
  with open('helper_functions.py','wb')as f:
    f.write(r.content)
  print("download finished")
#import that
from helper_functions import *
```
i used time.sleep() for a little fancy of seeing it downloads

#Greatlovefrombangladesh

**Reply by shisirkha** [reply]
@gaurav-soruppu Please reply that is it working or not.

i will try to help my best if it doesn't work or if you want to further understand what is cooking up in my code.
